hack/release.sh

```diff
@@ -0,0 +1,219 @@
+  if ! docker ps 1>/dev/null 2>&1 ; then
+    echo "Docker not available"
+    exit 1
+  fi
+}
+
+function build_images() {
+  if [ "${RELEASE}" ]; then
+    # On an annotated tag
+    MANAGER_IMAGE_NAME=${CAPV_MANAGER_IMAGE_RELEASE}
+    MANIFESTS_IMAGE_NAME=${CAPV_MANIFESTS_IMAGE_RELEASE}
+  else
+    MANAGER_IMAGE_NAME=${CAPV_MANAGER_IMAGE_CI}
+    MANIFESTS_IMAGE_NAME=${CAPV_MANIFESTS_IMAGE_CI}
+  fi
+  if [ "${PR}" ]; then
+    MANAGER_IMAGE_NAME=${CAPV_MANAGER_IMAGE_PR}
+    MANIFESTS_IMAGE_NAME=${CAPV_MANIFESTS_IMAGE_PR}
+  fi
+
+  # Manager image
+  echo "building" "${MANAGER_IMAGE_NAME}":"${VERSION}"
+  docker build \
+    -f Dockerfile \
+    -t "${MANAGER_IMAGE_NAME}":"${VERSION}" \
+    .
+  if [ "${LATEST}" ]; then
+    echo "tagging image" "${MANAGER_IMAGE_NAME}":"${VERSION}" "as latest"
+    docker tag "${MANAGER_IMAGE_NAME}":"${VERSION}" "${MANAGER_IMAGE_NAME}":latest
+  fi
+
+  # Manifests image
+  echo "building" "${MANIFESTS_IMAGE_NAME}":"${VERSION}"
+  docker build \
+    -f hack/tools/generate-yaml/Dockerfile \
+    -t "${MANIFESTS_IMAGE_NAME}":"${VERSION}" \
+    --build-arg "CAPV_MANAGER_IMAGE=${MANAGER_IMAGE_NAME}:${VERSION}" \
+    .
+  if [ "${LATEST}" ]; then
+    echo "tagging image" "${MANIFESTS_IMAGE_NAME}":"${VERSION}" "as latest"
+    docker tag "${MANIFESTS_IMAGE_NAME}":"${VERSION}" "${MANIFESTS_IMAGE_NAME}":latest
+  fi
+}
+
+function push_images() {
+  [ "${MANAGER_IMAGE_NAME}" ] || exit 1
+  [ "${MANIFESTS_IMAGE_NAME}" ] || exit 1
+  [ "${GCR_KEY_FILE}" ] || (echo "GCR_KEY_FILE not set"; exit 1)
+
+  docker login -u _json_key --password-stdin gcr.io <"${GCR_KEY_FILE}"
+
+  # Manager image
+  echo "pushing" "${MANAGER_IMAGE_NAME}":"${VERSION}"
+  docker push "${MANAGER_IMAGE_NAME}":"${VERSION}"
+  if [ "${LATEST}" ]; then
+    echo "also pushing" "${MANAGER_IMAGE_NAME}":"${VERSION}" "as latest"
+    docker push "${MANAGER_IMAGE_NAME}":latest
+  fi
+
+  # Manifests image
+  echo "pushing" "${MANIFESTS_IMAGE_NAME}":"${VERSION}"
+  docker push "${MANIFESTS_IMAGE_NAME}":"${VERSION}"
+  if [ "${LATEST}" ]; then
+    echo "also pushing" "${MANIFESTS_IMAGE_NAME}":"${VERSION}" "as latest"
+    docker push "${MANIFESTS_IMAGE_NAME}":latest
+  fi
+}
+
+function build_clusterctl() {
+  for os in linux darwin; do
+    # Don't build Darwin for PRs
+    if [ "${PR}" ] && [ "${os}" == "darwin" ]; then
+      continue
+    fi
+    echo "building clusterctl for ${os}"
+    GOOS="${os}" GOARCH=amd64 make clusterctl-in-docker
+  done
+}
+
+function push_clusterctl() {
+  local bucket="capv-ci"
+  if [ "${PR}" ]; then
+    bucket="capv-pr"
+  elif [ "${RELEASE}" ]; then
+    bucket="capv-release"
+  fi
+  for os in linux darwin; do
+    # Don't build Darwin for PRs
+    if [ "${PR}" ] && [ "${os}" == "darwin" ]; then
+      continue
+    fi
+    echo "copying clusterctl version ${VERSION} for ${os} to ${bucket}"
+    gsutil cp bin/clusterctl."${os}"_amd64 gs://"${bucket}"/"${VERSION}"/bin/"${os}"/amd64/clusterctl
```

Comment:
Hi @codenrhoden,

Do we want to add a shasum?

Reply:
Yeah, will a SHA256 sum.